eclair-core/src/main/scala/fr/acinq/eclair/crypto/TransportHandler.scala

```diff
@@ -178,13 +178,20 @@ class TransportHandler[T: ClassTag](keyPair: KeyPair, rs: Option[ByteVector], co
   when(Normal) {
     handleExceptions {
       case Event(Tcp.Received(data), d: NormalData[T @unchecked]) =>
+        log.debug("received chunk of size={}", data.size)
         val (dec1, plaintextMessages) = d.decryptor.copy(buffer = d.decryptor.buffer ++ data).decrypt()
         if (plaintextMessages.isEmpty) {
           connection ! Tcp.ResumeReading
           stay() using d.copy(decryptor = dec1)
         } else {
-          log.debug("read {} messages, waiting for readacks", plaintextMessages.size)
+          log.debug("decoding {} raw messages", plaintextMessages.size)
           val unackedReceived = sendToListener(d.listener, plaintextMessages)
+          if (unackedReceived.isEmpty) {
+            log.debug("no decoded messages in this chunk, resuming reading")
+            connection ! Tcp.ResumeReading
```

Comment:
It's probably worth a comment explaining that when there are known messages in this chunk, we let the `PeerConnection` decide when to do a `ReadAck` which will then resume reading, that's why in the case where there's nothing forwarded to the `PeerConnection`, we need to do it ourselves here.

Reply:
Done with ff30986ce, but I think e998fe5 makes the logic obvious: whenever `unackedReceived` is empty, we call `Tcp.ResumeReading`.